dotcom-rendering/src/lib/getDataLinkName.ts

```diff
@@ -1,9 +1,15 @@
 import { ArticleDesign, ArticleSpecial } from '@guardian/libs';
+import type { FEFrontCardStyle } from '../types/front';
 
 /**
- * TODO: missing "podcast" and "external"
+ * TODO: missing "podcast"
  */
-const getLinkType = ({ theme, design }: ArticleFormat): RichLinkCardType => {
+const getLinkType = (
+	{ theme, design }: ArticleFormat,
+	cardStyle?: FEFrontCardStyle,
+): RichLinkCardType => {
+	if (cardStyle === 'ExternalLink') return 'external';
```

Comment:
Frontend flow:

[`facia-press`](https://github.com/guardian/frontend/blob/main/common/app/model/CardStyle.scala#L24) calls [`facia-client`](https://github.com/guardian/facia-scala-client/blob/main/fapi-client/src/main/scala/com/gu/facia/api/utils/CardStyle.scala#L29) in order to decide the `CardStyle`.

On the rendering layer [`facia` decides the `data-link-name`](https://github.com/guardian/frontend/blob/main/common/app/views/fragments/items/facia_cards/contentCard.scala.html#LL25C1-L26C1) based on the `CardStyle`:

* https://github.com/guardian/frontend/blob/main/common/app/layout/FaciaCard.scala#L148-L151
* https://github.com/guardian/frontend/blob/main/common/app/layout/FaciaCard.scala#L201